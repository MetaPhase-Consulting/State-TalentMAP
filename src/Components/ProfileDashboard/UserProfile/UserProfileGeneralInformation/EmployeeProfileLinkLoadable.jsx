import createLoader from '../../../Loadable';

export const path = () => import('./EmployeeProfileLink');

const Loadable = createLoader({ path, shouldPreload: false, usePlaceholder: false });

const EPLLoadable = ({ ...rest }) => (
  <Loadable {...rest} />
);

export default EPLLoadable;
