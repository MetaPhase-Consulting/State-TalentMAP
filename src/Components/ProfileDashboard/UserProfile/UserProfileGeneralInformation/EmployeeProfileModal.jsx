import PropTypes from 'prop-types';
import swal from '@sweetalert/with-react';
import { Viewer, Worker } from '@react-pdf-viewer/core';
import { fetchJWT } from 'utilities';
import { defaultLayoutPlugin } from '@react-pdf-viewer/default-layout';

// Import styles
import '@react-pdf-viewer/core/lib/styles/index.css';
import '@react-pdf-viewer/default-layout/lib/styles/index.css';

const ChecklistModal = props => {
  const { url } = props;

  const cancel = (e) => {
    e.preventDefault();
    try {
      swal.close();
    } catch {
      return null;
    }
    return null;
  };

  const defaultLayoutPluginInstance = defaultLayoutPlugin();

  return (
    <div className="checklist-modal">
      <div>
        <Worker workerUrl="/static/js/bundle.worker.js">
          {/* <Viewer fileUrl={this.state.data} />; */}
          <Viewer
            fileUrl={url}
            httpHeaders={{ JWTAuthorization: fetchJWT() }}
            withCredentials
            plugins={[defaultLayoutPluginInstance]}
          />
        </Worker>
        <div className="checklist-modal-buttons-container">
          <button type="button" onClick={cancel}>Close</button>
        </div>
      </div>
    </div>
  );
};

ChecklistModal.propTypes = {
  url: PropTypes.string,
};

ChecklistModal.defaultProps = {
  url: '',
};

export default ChecklistModal;
