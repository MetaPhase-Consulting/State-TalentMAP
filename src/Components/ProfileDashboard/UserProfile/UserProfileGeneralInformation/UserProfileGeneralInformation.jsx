import React from 'react';
import PropTypes from 'prop-types';
import { get } from 'lodash';
import { NO_GRADE, NO_SKILL } from 'Constants/SystemMessages';
import { USER_PROFILE } from 'Constants/PropTypes';
import SectionTitle from '../../SectionTitle';
import InformationDataPoint from '../../InformationDataPoint';
import EditProfile from '../EditProfile';
import Avatar from '../../../Avatar';
import StaticDevContent from '../../../StaticDevContent';
import SkillCodeList from '../../../SkillCodeList';

const UserProfileGeneralInformation = ({ userProfile, showEditLink, useGroup, isPublic }) => {
  const avatar = {
    firstName: get(userProfile, 'user.first_name'),
    lastName: get(userProfile, 'user.last_name'),
    initials: userProfile.initials,
    displayName: userProfile.display_name,
    externalSource: get(userProfile, 'avatar'),
    externalSourceToUse: 'm',
  };
  const infoDataPointClassName = 'skill-code-data-point-container skill-code-data-point-container-gen-spec';
  const conditionalStaticDevContent = isPublic ?
    (<InformationDataPoint
      content={`Grade: ${get(userProfile, 'grade', NO_GRADE)}`}
      className={infoDataPointClassName}
    />)
    :
    (<InformationDataPoint
      content={`${get(userProfile, 'user.skill', NO_SKILL)} â€¢ ${get(userProfile, 'user.grade', NO_GRADE)}`}
      className="skill-code-data-point-container skill-code-data-point-container-gen-spec"
    />);
  return (
    <div className="current-user-top current-user-section-border current-user-section-container">
      <div className="section-padded-inner-container">
        <div className="avatar-group">
          <Avatar
            className="dashboard-user-profile-picture"
            {...avatar}
          />
        </div>
        { showEditLink && <StaticDevContent><EditProfile /></StaticDevContent> }
        <div className="name-group">
          <SectionTitle small title={`${userProfile.user.last_name ? `${userProfile.user.last_name}, ` : ''}${userProfile.user.first_name}`} className="current-user-name" />
          {conditionalStaticDevContent}
          {
            !useGroup &&
              <InformationDataPoint
                content={<SkillCodeList skillCodes={userProfile.skills} />}
                className="skill-code-data-point-container skill-code-data-point-container-skill"
              />
          }
        </div>
      </div>
    </div>
  );
};

UserProfileGeneralInformation.propTypes = {
  userProfile: USER_PROFILE.isRequired,
  showEditLink: PropTypes.bool,
  useGroup: PropTypes.bool,
  isPublic: PropTypes.bool,
};

UserProfileGeneralInformation.defaultProps = {
  showEditLink: true,
  useGroup: false,
  isPublic: false,
};

export default UserProfileGeneralInformation;
