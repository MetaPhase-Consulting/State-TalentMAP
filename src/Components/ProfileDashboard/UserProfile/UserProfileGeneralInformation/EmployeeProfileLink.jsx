import { Component } from 'react';
import swal from '@sweetalert/with-react';
import { USER_PROFILE } from 'Constants/PropTypes';
import InteractiveElement from 'Components/InteractiveElement';
import { isOnProxy } from 'utilities';
import InformationDataPoint from '../../InformationDataPoint';
import EmployeeProfileModal from './EmployeeProfileModal';

class EmployeeProfileLink extends Component {
  constructor(props) {
    super(props);
    this.state = {
    };
  }
  render() {
    const { userProfile } = this.props;
    const emp_profile_urls = userProfile?.employee_profile_url;

    let url$ = emp_profile_urls?.internal || emp_profile_urls?.internalRedacted;
    if (isOnProxy()) {
      url$ = emp_profile_urls?.external || emp_profile_urls?.externalRedacted;
    }

    const openPdf = () => swal({
      title: 'Employee Profile Report:',
      button: false,
      className: 'modal-1300',
      content: (
        <EmployeeProfileModal
          url={url$}
        />
      ),
    });

    return (
      <InformationDataPoint
        content={
          <InteractiveElement
            onClick={openPdf}
            type="a"
            title="Download Employee Profile PDF"
          >
            Employee Profile
          </InteractiveElement>
        }
      />
    );
  }
}

EmployeeProfileLink.propTypes = {
  userProfile: USER_PROFILE.isRequired,
};

export default EmployeeProfileLink;
