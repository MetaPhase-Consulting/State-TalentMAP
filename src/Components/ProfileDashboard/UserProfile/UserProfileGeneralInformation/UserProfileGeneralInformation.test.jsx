import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { bidderUserObject } from '../../../../__mocks__/userObject';
import UserProfileGeneralInformation from './UserProfileGeneralInformation';

describe('UserProfileGeneralInformationComponent', () => {
  it('is defined', () => {
    const wrapper = shallow(<UserProfileGeneralInformation.WrappedComponent
      userProfile={bidderUserObject}
    />);
    expect(wrapper).toBeDefined();
  });

  it('matches snapshot when showEditLink is true', () => {
    const wrapper = shallow(
      <UserProfileGeneralInformation.WrappedComponent
        showEditLink
        userProfile={bidderUserObject}
      />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('matches snapshot when useGroup is true', () => {
    const wrapper = shallow(
      <UserProfileGeneralInformation.WrappedComponent
        useGroup
        userProfile={bidderUserObject}
      />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('matches snapshot', () => {
    const wrapper = shallow(<UserProfileGeneralInformation.WrappedComponent
      userProfile={bidderUserObject}
    />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
