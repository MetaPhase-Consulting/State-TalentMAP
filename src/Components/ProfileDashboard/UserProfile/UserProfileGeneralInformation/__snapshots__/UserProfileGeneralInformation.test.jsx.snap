// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProfileGeneralInformationComponent matches snapshot 1`] = `
<div
  className="current-user-top current-user-section-border current-user-section-container"
>
  <div
    className="section-padded-inner-container"
  >
    <MediaQueryWrapper
      breakpoint="screenXlgMin"
      widthType="min"
    >
      <Component />
    </MediaQueryWrapper>
    <div
      className="name-group"
    >
      <PositionInformation
        className="current-user-name"
        icon=""
        small={true}
        title="Doe, John"
      />
      <EPLLoadable
        userProfile={
          Object {
            "bid_statistics": Array [
              Object {
                "draft": 1,
                "submitted": 1,
              },
            ],
            "cdo": Object {
              "email": "shadtrachl@state.gov",
              "first_name": "Leah",
              "initials": "LS",
              "last_name": "Shadtrach",
              "username": "shadtrachl",
            },
            "display_name": "John",
            "emp_id": "1",
            "employee_info": Object {
              "grade": "00",
              "skills": Array [
                Object {
                  "code": "6218",
                  "description": "CONSTRUCTION ENGINEERING",
                },
                Object {
                  "code": "6211",
                  "description": "COMPUTER SCIENCE",
                },
              ],
            },
            "employee_profile_url": Object {
              "external": "https://external.com",
              "internal": "https://internal.com",
            },
            "favorite_positions": Array [
              Object {
                "id": 1,
                "representation": "[00003026] OMS (COM) (Freetown, Sierra Leone)",
              },
              Object {
                "id": 4,
                "representation": "[00180000] OMS (DCM) (Addis Ababa, Ethiopia)",
              },
            ],
            "favorite_positions_pv": Array [
              Object {
                "id": 10,
              },
              Object {
                "id": 20,
              },
            ],
            "favorite_tandem_positions": Array [
              Object {
                "id": 1,
                "representation": "[00003026] OMS (COM) (Freetown, Sierra Leone)",
              },
              Object {
                "id": 4,
                "representation": "[00180000] OMS (DCM) (Addis Ababa, Ethiopia)",
              },
            ],
            "favorite_tandem_positions_pv": Array [
              Object {
                "id": 10,
              },
              Object {
                "id": 20,
              },
            ],
            "grade": "03",
            "id": 1,
            "initials": "JD",
            "is_cdo": false,
            "skills": Array [
              63,
              40,
            ],
            "user": Object {
              "email": "doej@state.gov",
              "first_name": "John",
              "initials": "JD",
              "last_name": "Doe",
              "username": "jdoe",
            },
            "user_info": Object {
              "office_address": "1234 Apple St. NW Washington, DC 20010",
              "office_phone": "202-876-5432",
            },
          }
        }
      />
      <InformationDataPoint
        className="skill-code-data-point-container skill-code-data-point-container-gen-spec"
        content="Grade: 00"
        sideBySide={false}
        title=""
        titleOnBottom={false}
      />
      <InformationDataPoint
        className="skill-code-data-point-container skill-code-data-point-container-skill"
        content={
          <SkillCodeList
            displayCodeFirst={false}
            skillCodes={
              Array [
                Object {
                  "code": "6218",
                  "description": "CONSTRUCTION ENGINEERING",
                },
                Object {
                  "code": "6211",
                  "description": "COMPUTER SCIENCE",
                },
              ]
            }
          />
        }
        sideBySide={false}
        title=""
        titleOnBottom={false}
      />
    </div>
  </div>
</div>
`;

exports[`UserProfileGeneralInformationComponent matches snapshot when showEditLink is true 1`] = `
<div
  className="current-user-top current-user-section-border current-user-section-container"
>
  <div
    className="section-padded-inner-container"
  >
    <MediaQueryWrapper
      breakpoint="screenXlgMin"
      widthType="min"
    >
      <Component />
    </MediaQueryWrapper>
    <div
      className="name-group"
    >
      <PositionInformation
        className="current-user-name"
        icon=""
        small={true}
        title="Doe, John"
      />
      <EPLLoadable
        userProfile={
          Object {
            "bid_statistics": Array [
              Object {
                "draft": 1,
                "submitted": 1,
              },
            ],
            "cdo": Object {
              "email": "shadtrachl@state.gov",
              "first_name": "Leah",
              "initials": "LS",
              "last_name": "Shadtrach",
              "username": "shadtrachl",
            },
            "display_name": "John",
            "emp_id": "1",
            "employee_info": Object {
              "grade": "00",
              "skills": Array [
                Object {
                  "code": "6218",
                  "description": "CONSTRUCTION ENGINEERING",
                },
                Object {
                  "code": "6211",
                  "description": "COMPUTER SCIENCE",
                },
              ],
            },
            "employee_profile_url": Object {
              "external": "https://external.com",
              "internal": "https://internal.com",
            },
            "favorite_positions": Array [
              Object {
                "id": 1,
                "representation": "[00003026] OMS (COM) (Freetown, Sierra Leone)",
              },
              Object {
                "id": 4,
                "representation": "[00180000] OMS (DCM) (Addis Ababa, Ethiopia)",
              },
            ],
            "favorite_positions_pv": Array [
              Object {
                "id": 10,
              },
              Object {
                "id": 20,
              },
            ],
            "favorite_tandem_positions": Array [
              Object {
                "id": 1,
                "representation": "[00003026] OMS (COM) (Freetown, Sierra Leone)",
              },
              Object {
                "id": 4,
                "representation": "[00180000] OMS (DCM) (Addis Ababa, Ethiopia)",
              },
            ],
            "favorite_tandem_positions_pv": Array [
              Object {
                "id": 10,
              },
              Object {
                "id": 20,
              },
            ],
            "grade": "03",
            "id": 1,
            "initials": "JD",
            "is_cdo": false,
            "skills": Array [
              63,
              40,
            ],
            "user": Object {
              "email": "doej@state.gov",
              "first_name": "John",
              "initials": "JD",
              "last_name": "Doe",
              "username": "jdoe",
            },
            "user_info": Object {
              "office_address": "1234 Apple St. NW Washington, DC 20010",
              "office_phone": "202-876-5432",
            },
          }
        }
      />
      <InformationDataPoint
        className="skill-code-data-point-container skill-code-data-point-container-gen-spec"
        content="Grade: 00"
        sideBySide={false}
        title=""
        titleOnBottom={false}
      />
      <InformationDataPoint
        className="skill-code-data-point-container skill-code-data-point-container-skill"
        content={
          <SkillCodeList
            displayCodeFirst={false}
            skillCodes={
              Array [
                Object {
                  "code": "6218",
                  "description": "CONSTRUCTION ENGINEERING",
                },
                Object {
                  "code": "6211",
                  "description": "COMPUTER SCIENCE",
                },
              ]
            }
          />
        }
        sideBySide={false}
        title=""
        titleOnBottom={false}
      />
    </div>
  </div>
</div>
`;

exports[`UserProfileGeneralInformationComponent matches snapshot when useGroup is true 1`] = `
<div
  className="current-user-top current-user-section-border current-user-section-container"
>
  <div
    className="section-padded-inner-container"
  >
    <MediaQueryWrapper
      breakpoint="screenXlgMin"
      widthType="min"
    >
      <Component />
    </MediaQueryWrapper>
    <div
      className="name-group"
    >
      <PositionInformation
        className="current-user-name"
        icon=""
        small={true}
        title="Doe, John"
      />
      <EPLLoadable
        userProfile={
          Object {
            "bid_statistics": Array [
              Object {
                "draft": 1,
                "submitted": 1,
              },
            ],
            "cdo": Object {
              "email": "shadtrachl@state.gov",
              "first_name": "Leah",
              "initials": "LS",
              "last_name": "Shadtrach",
              "username": "shadtrachl",
            },
            "display_name": "John",
            "emp_id": "1",
            "employee_info": Object {
              "grade": "00",
              "skills": Array [
                Object {
                  "code": "6218",
                  "description": "CONSTRUCTION ENGINEERING",
                },
                Object {
                  "code": "6211",
                  "description": "COMPUTER SCIENCE",
                },
              ],
            },
            "employee_profile_url": Object {
              "external": "https://external.com",
              "internal": "https://internal.com",
            },
            "favorite_positions": Array [
              Object {
                "id": 1,
                "representation": "[00003026] OMS (COM) (Freetown, Sierra Leone)",
              },
              Object {
                "id": 4,
                "representation": "[00180000] OMS (DCM) (Addis Ababa, Ethiopia)",
              },
            ],
            "favorite_positions_pv": Array [
              Object {
                "id": 10,
              },
              Object {
                "id": 20,
              },
            ],
            "favorite_tandem_positions": Array [
              Object {
                "id": 1,
                "representation": "[00003026] OMS (COM) (Freetown, Sierra Leone)",
              },
              Object {
                "id": 4,
                "representation": "[00180000] OMS (DCM) (Addis Ababa, Ethiopia)",
              },
            ],
            "favorite_tandem_positions_pv": Array [
              Object {
                "id": 10,
              },
              Object {
                "id": 20,
              },
            ],
            "grade": "03",
            "id": 1,
            "initials": "JD",
            "is_cdo": false,
            "skills": Array [
              63,
              40,
            ],
            "user": Object {
              "email": "doej@state.gov",
              "first_name": "John",
              "initials": "JD",
              "last_name": "Doe",
              "username": "jdoe",
            },
            "user_info": Object {
              "office_address": "1234 Apple St. NW Washington, DC 20010",
              "office_phone": "202-876-5432",
            },
          }
        }
      />
      <InformationDataPoint
        className="skill-code-data-point-container skill-code-data-point-container-gen-spec"
        content="Grade: 00"
        sideBySide={false}
        title=""
        titleOnBottom={false}
      />
      <InformationDataPoint
        className="skill-code-data-point-container skill-code-data-point-container-skill"
        content={
          <SkillCodeList
            displayCodeFirst={false}
            skillCodes={
              Array [
                Object {
                  "code": "6218",
                  "description": "CONSTRUCTION ENGINEERING",
                },
                Object {
                  "code": "6211",
                  "description": "COMPUTER SCIENCE",
                },
              ]
            }
          />
        }
        sideBySide={false}
        title=""
        titleOnBottom={false}
      />
    </div>
  </div>
</div>
`;
